music_assistant/providers/spotify/helpers.py

```diff
@@ -1,32 +1,200 @@
-"""Helpers/utils for the Spotify musicprovider."""
-
-from __future__ import annotations
-
-import os
-import platform
-
-from music_assistant.helpers.process import check_output
-
-
-async def get_librespot_binary() -> str:
-    """Find the correct librespot binary belonging to the platform."""
-
-    async def check_librespot(librespot_path: str) -> str | None:
-        try:
-            returncode, output = await check_output(librespot_path, "--version")
-            if returncode == 0 and b"librespot" in output:
-                return librespot_path
-        except OSError:
-            return None
-
-    base_path = os.path.join(os.path.dirname(__file__), "bin")
-    system = platform.system().lower().replace("darwin", "macos")
-    architecture = platform.machine().lower()
-
-    if bridge_binary := await check_librespot(
-        os.path.join(base_path, f"librespot-{system}-{architecture}")
-    ):
-        return bridge_binary
-
-    msg = f"Unable to locate Librespot for {system}/{architecture}"
-    raise RuntimeError(msg)
+"""Helper functions for Spotify provider."""
+
+from __future__ import annotations
+
+import os
+import platform
+from typing import TYPE_CHECKING, Any, cast
+from urllib.parse import urlencode
+
+import pkce
+from music_assistant_models.config_entries import ConfigEntry
+from music_assistant_models.enums import ConfigEntryType
+
+from music_assistant.helpers.app_vars import app_var  # type: ignore[attr-defined]
+from music_assistant.helpers.auth import AuthenticationHelper
+from music_assistant.helpers.process import check_output
+
+from .constants import (
+    CALLBACK_REDIRECT_URL,
+    CONF_ACTION_AUTH,
+    CONF_ACTION_CLEAR_AUTH,
+    CONF_CLIENT_ID,
+    CONF_ENABLE_PODCASTS,
+    CONF_PLAYED_THRESHOLD,
+    CONF_REFRESH_TOKEN,
+    CONF_SYNC_PLAYED_STATUS,
+    SCOPE,
+)
+
+if TYPE_CHECKING:
+    from music_assistant import MusicAssistant
+
+
+async def get_librespot_binary() -> str:
+    """Find the correct librespot binary belonging to the platform."""
+
+    async def check_librespot(librespot_path: str) -> str | None:
+        try:
+            returncode, output = await check_output(librespot_path, "--version")
+            if returncode == 0 and b"librespot" in output:
+                return librespot_path
+            return None
+        except OSError:
+            return None
+
+    base_path = os.path.join(os.path.dirname(__file__), "bin")
+    system = platform.system().lower().replace("darwin", "macos")
+    architecture = platform.machine().lower()
+
+    if bridge_binary := await check_librespot(

```

Comment:
bridge_binary ?

Reply:
That is the variable name in the existing code?